MainModule/Client/UI/Default/YesNoPrompt.lua

```diff
@@ -45,19 +46,36 @@ return function(data)
 		BackgroundColor3 = Color3.fromRGB(206, 72, 45);
 		BackgroundTransparency = 0.5;
 	})
-	
+
 	yes.MouseButton1Down:Connect(function()
+		if data.Delay and os.clock() < startTick + data.Delay then
+			return
+		end
+
 		answer = "Yes"
 		window:Close()
 	end)
-	
+
 	no.MouseButton1Down:Connect(function()
 		answer = "No"
 		window:Close()
 	end)
-	
+
 	gTable = window.gTable
 	window:Ready()
-	repeat wait() until answer 
+
+	if data.Delay then
+		yes.BackgroundColor3 = Color3.fromRGB(38, 100, 28)
+
+		repeat
+			yes.Text = string.format("%s seconds left before\nclicking allowed", math.ceil(startTick + data.Delay - os.clock()))
```

Comment:
This would probably be better as just the number counting down as it's simpler and still gets the point across

Reply:
Done.